all/pom.xml

```diff
@@ -187,5 +192,9 @@
             <version>${project.version}</version>
             <type>zip</type>
         </dependency>
+        <dependency>
+            <groupId>com.adobe.cq</groupId>
+            <artifactId>core.wcm.components.core</artifactId>
```

Comment:
oh we have now a hard dependency on core components, right? because of the interface & utility function? 
I am just bit worried because this can conflict with a client core component version...

Reply:
let's discuss this in some call

Reply:
@igorsimovski : how much code would you need to clone of we don't add the dependency?

I'd like to avoid adding the dependency to core competents to not dictate how and which version needs to be used by the client

Reply:
Just a thought: can't this be overcome with a specific entry in the **Import-Package** section in the client project?

Reply:
> @igorsimovski : how much code would you need to clone of we don't add the dependency?
com.adobe.cq.wcm.core.components.models.datalayer.ComponentData (we use the getData method from this interface)
com.adobe.cq.wcm.core.components.internal.jackson.ComponentDataModelSerializer (this is the serializer for the field/'json property' associated with getData())
com.adobe.cq.wcm.core.components.util.ComponentUtils (to check if data layer is enabled in CA configs - This can be omitted either by making our own ca config or rely only on a dataLayer on/off switch in the components)
com.adobe.cq.wcm.core.components.models.datalayer.builder.DataLayerBuilder (this is a rather large implementation that offers a convenient way of building the dataLayer json, but we could do w/o it - we can simply do it with a dedicated POJO for each component)

Reply:
> Just a thought: can't this be overcome with a specific entry in the **Import-Package** section in the client project?

https://www.baeldung.com/maven-version-collision

Reply:
Sorry for my late reply. I think it makes sense to discuss this one to one